src/zenml/pipelines/base_pipeline.py

```diff
@@ -153,14 +154,24 @@ def _verify_arguments(self, *steps: BaseStep, **kw_steps: BaseStep) -> None:
         combined_steps = {}
         step_classes: Dict[Type[BaseStep], str] = {}
 
-        for i, step in enumerate(steps):
+        def _verify_step(key: str, step: BaseStep) -> None:
+            """Verifies a single step of the pipeline."""
             step_class = type(step)
-            key = input_step_keys[i]
+
+            if isinstance(step, BaseStepMeta):
+                raise PipelineInterfaceError(
+                    f"Wrong argument type (`{step_class}`) for argument "
+                    f"'{key}' of pipeline '{self.name}'. "
+                    f"A `BaseStep` subclass was provided instead of an instance. "
+                    f"This might have been caused by missing brackets when "
```

Comment:
It's better message than before, but instead of `This might have been caused by missing brackets` you can think of writing `This might have caused due to missing brackets of your steps`

Reply:
Great suggestion! Changed the message accordingly.